connector/connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -81,6 +81,31 @@ class SparkConnectProtoSuite extends PlanTest with SparkConnectPlanTest {
     }
   }
 
+  test("Test union, except, intersect") {
+    for (isAll <- Seq(true, false)) {
+      val connectPlan = {
+        import org.apache.spark.sql.connect.dsl.plans._
+        transform(connectTestRelation.except(connectTestRelation, isAll))
+      }
+      val sparkPlan = sparkTestRelation.except(sparkTestRelation, isAll)
+      comparePlans(connectPlan.analyze, sparkPlan.analyze, false)
+
+      val connectPlan2 = {
+        import org.apache.spark.sql.connect.dsl.plans._
+        transform(connectTestRelation.intersect(connectTestRelation, isAll))
+      }
+      val sparkPlan2 = sparkTestRelation.intersect(sparkTestRelation, isAll)
+      comparePlans(connectPlan2.analyze, sparkPlan2.analyze, false)
+    }
+
+    val connectPlan3 = {
+      import org.apache.spark.sql.connect.dsl.plans._
+      transform(connectTestRelation.union(connectTestRelation))
```

Comment:
last question: does the current spark connect client API support specifying `isAll` for Union?

Reply:
My understanding is no:

there are 3 unions: `unionAll`, `unionByName` and `union distinct`. For the third case DataFrame API says to use `DISTINCT(UNION(....))`. For the `UNIONAll` itself there is no flag for `isAll` so by default `Union` is `UnionAll`

There should be some context for this difference but I cannot tell why now.

Reply:
This is one of the cases where we shouldn't just blindly take what the DataFrame API does but we should think in terms of generalized parse plans. 


In this case I much rather have the option on the union for a All rather than doing something implicit.

Reply:
We are doing Union with isAll because along with intersect and except in the same message.

I don't agree with it though. Generally speaking if there is something DataFrame decides to do, I think figuring out the context will be useful. People working on Spark are smart. They can of course just does union with isAll but they decided to not do so. There must be a valid reason and that reason might also apply to connect.

Reply:
Another reason actually to follow Catalyst: our clients, by best effort, should follow existing data frame API to achieve changing imports and rest of code work. 

So it's expected that users will use union().distinct() to construct the case above. If we ask clients to smartly identify these two API can be coalesce together, that will be too much for clients. Instead clients just convert each API to a plan node and connect proto follow Catalyst then it will work.

Reply:
Spark Connect may have many clients, built-in or third-party. I think it's better to make the query plan proto definition general/standard.

It was kind of a mistake that we forgot to add DataFrame APIs for users to specify `ALL | DISTINCT` for set operations. We added the `isAll` flag to the plan later, because the SQL parser started to support it.

`byName` flag is only supported by Union now, but I think other two set operators should support it as well.

To summarize by proposal:
1. add a `byName` flag to `SetOperation` proto definition
2. The server translate `Union(..., isAll = false)` to `Distinct(Union)`
3. The server fails if `byName = true` and the set type is intercept or except.

Reply:
Generally, the proto definitions should follow as much as possible the basics for the relational algebra / SQL language. As @cloud-fan mentioned, since there maybe other clients. It's important to not just follow the quirks of the DF API because its the way it's currently implemented.

Having a proper `SetOperation` definition is not stopping us from continuing to have customers call `df.union().distinct()`.

Reply:
I am fine on the proposal as long as it does not prevent the clients being compatible with existing data frame API. Any extra functionality is fine.

Thanks @cloud-fan for you summary. Let me change this PR.

Reply:
Hm, sorry for joining this conversation late. Why don't we just match it to SQL plan implementation? If the design is the issue, we can refactor them together?

Reply:
The issue is, we can't change proto message freely. It's API, while catalyst plans are internal implementations. That's why we need more considerations when adding proto messages, to make them future-proof.